### ISSUE TITLE
Hide create new option prompt text when input is empty/reset/cleared

### ISSUE THREAD

**Issue by kukikiloke** [JedWatson/react-select#1575]
Preconditions:
- multi: true
- createable: true

Steps:
1. Type something which will show some of the options, select any of the options.
2. Input is cleared on selecting option but the prompt text `Create an option "xxx"` is still there when you focus on the field. The same happens when you type something and click outside of the select container (blur) and focus on it again.

It would be great if the prompt text `Create an option "xxx"` will not display when the input is empty/reset/cleared as it's kinda strange to display suggestion to create new option when there is nothing in the input. Thanks!

**Comment by slavab89** [comment]
+1

**Comment by shoaibbhimani** [comment]
+1

**Comment by shoaibbhimani** [comment]
```
I found workaround which might help any one looking to do the same 

1) I used option optionRenderer to render Show Menu List and added classes to Show or Hide based on this.showMenu Flag follow code might help 

2) I also needed to changes some css,  remove padding from following classes 
 
.Select-option {
  padding:0;
}

.Select-noresults {
  padding:0;
}

====================
 Add padding to this class which shows Menu list to look a bit nicer like before

.visible {
 padding: 8px 10px;
}

.hide {
 display:none;
}


class InputBox extends Component {
    constructor(props){
      super(props);
      this.showMenu = true;
    }

 loadOptions = (searchTerm, callback) => {
  if (!searchTerm) {
    return Promise.resolve({ options: [] });
  }

   $.ajax({
     url:'url'',
     type:'get',
     data: {
         q: searchTerm
     },
     headers:{
       'Content-Type': 'application/json',
       'X-Requested-With': 'XMLHttpRequest',
        'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
     },
     success:(data) => {
       callback(null, {
         options: data
       });
     },
     error:() => {
       console.log("Err")
     },
   });
 }


    render(){
      const AsyncComponent = creatable
          ? Select.AsyncCreatable
          : Select.Async;

      return (
       <section>
         <AsyncComponent
           multi={true}
           value={value}
           onChange={onChange}
           valueKey={valueKey}
           labelKey={labelKey}
         // To Detect when User is Searching because we cant use onInputChange as it do not work in Async Select Option  
         loadOptions={(searchTerm, callback) => {
            // this Flag causes menu list to show up Again
             this.showMenu = true;
             return this.loadOptions(searchTerm, callback);
           }}
         // this line is to Remove Create option "xxx" from Menu 
           promptTextCreator={(label) => label + '   ' }
           onFocus={() => {
             // onFocus Menu Hides Menu list
             this.showMenu = false;
            }}
           optionRenderer={(name) => {
             return (
              // visible and hide class shows and hide menu list class code is listing above
               <li className={this.showMenu ? 'visible' : 'hide'}>
                 {name}
               </li>
             )
           }} 
        />
```

**Comment by tonisostrat** [comment]
A quick (and dirty?) workaround currently is to manually delete the internal placeholder item whenever `onChange()` fires.

Here's an example using a stateless component:

```javascript
export default props => {
  let ref;

  return (<Select.Creatable ref={el => ref = el}
                            onChange(values => {
                              delete ref._createPlaceholderOption;
                              //resume handling the values
                            }/>)
}



```

**Comment by shoaibbhimani** [comment]
@tonisostrat  why not simply hide element based on its css class 
.Select-menu-outer .Select-create-option-placeholder {
  display: none;
}

**Comment by tonisostrat** [comment]
Because that only hides it visually. It's still persisted inside the ``<Select/>` and may affect the behaviour of the whole field in regards to filtering the drop-down and item creation/selection.

**Comment by batmandarkside** [comment]
+1

**Comment by thejjmiller** [comment]
+1

**Comment by lorenzos** [comment]
+1!

@tonisostrat Your workaround doesn't work for me :(

**Comment by lorenzos** [comment]
To me, this is fixed with #1306. Close?

**Comment by bladey** [comment]
Hello -

In an effort to sustain the `react-select` project going forward, we're closing old issues / pull requests.

We understand this might be inconvenient but in the best interest of supporting the broader community we have to direct our limited efforts to maintain the latest version.

If you feel this issue / pull request is still relevant and you'd like us to review it, please leave a comment and we'll do our best to get back to you.
